source_gen/lib/src/builder.dart

```diff
@@ -144,24 +152,57 @@ class _Builder extends Builder {
   }
 }
 
+/// A [Builder] which generates `part of` files.
+///
+/// Generated files will be prefixed with a `partId` to ensure multiple
+/// [SharedPartBuilder]s can produce non conflicting `part of` files.
+class SharedPartBuilder extends _Builder {
+  /// Wrap [generators] as a [Builder] that generates `part of` files.
+  ///
+  /// [partId] indicates what files will be created for each `.dart`
+  /// input. This extension should be unique as to not conflict with other
+  /// [SharedPartBuilder]s. The resulting file will be of the form
+  /// `<generatedExtension>.g.part`. If any generator in [generators] will
+  /// create additional outputs through the [BuildStep] they should be indicated
+  /// in [additionalOutputExtensions].
+  ///
+  /// [formatOutput] is called to format the generated code. Defaults to
+  /// [DartFormatter.format].
+  ///
+  /// [header] is used to specify the content at the top of each generated file.
+  /// If `null`, the content of [defaultFileHeader] is used.
+  /// If [header] is an empty `String` no header is added.
+  SharedPartBuilder(List<Generator> generators, String partId,
+      {String formatOutput(String code),
+      List<String> additionalOutputExtensions = const [],
+      String header})
```

Comment:
I don't think header is something we can/want to support with this

@kevmoo - are you OK if we drop this?

Reply:
We need to keep it – it allows us to support copyright headers, `// ignore_for_file`, etc...

Reply:
Move it to the combining builder?

Reply:
Do we have any use cases where a builder itself is configuring this? I thought this was always configured by the user via build.yaml.